### ISSUE TITLE
Assertion `ct.predictionTime > lastSendTime' failed.

### ISSUE THREAD

**Issue by sebwolf-de** [SeisSol/SeisSol#776]
**Describe the bug**
When I start a SeisSol in Debug mode, the initialization runs smoothly. Once the actual simulation starts, I get the following error:
```
SeisSol_Debug_dskx_6_elastic: /dss/dsshome1/03/ga35kum3/SeisSol/master/src/Solver/time_stepping/GhostTimeCluster.cpp:26: void seissol::time_stepping::GhostTimeCluster::receiveGhostLayer(): Assertion `ct.predictionTime > lastSendTime' failed.
```

**Expected behavior**
SeisSol runs the simulation phase, succesfully

**To Reproduce**
1. Which version do you use? Provide branch and commit id.
Latest Master: `4889c3a0`
2. Which build settings do you use? Which compiler version do you use? 
Intel Compiler. I assume this assertion fails in all debug configurations, e.g. the one below. It has failed with other debug builds.
![Screenshot from 2023-01-18 10-35-31](https://user-images.githubusercontent.com/24592027/213135819-6f33cb8f-2787-4157-b636-7ea66b8d0c8c.png)
3. On which machine does your problem occur? If on a cluster: Which modules are loaded?
SuperMUC-NG:
```
Currently Loaded Modulefiles:
admin/1.0
tempdir/1.0
lrz/1.0
spack/22.2.1
intel-oneapi-compilers/2021.4.0
intel-oneapi-mkl/2022.0.2
intel-oneapi-mpi/2021.5.1-intel
gcc/8.5.0
cmake/3.21.4
eigen/3.4.0-intel21
netcdf-hdf5-all/4.6_hdf5-1.10-intel21-impi
numactl/2.0.14-intel21
python/3.6_intel
yaml-cpp/0.7.0-intel21
```
4. Provide parameter/material files.
LOH1 setup as it is in the Examples Repo, run on 4 nodes.

**Screenshots/Console output**
```
Wed Jan 18 10:31:04, Info:  Cleaning possible double occurring receivers for MPI...
Wed Jan 18 10:31:04, Info:  Mapping receivers to LTS cells...
Rank:        0 | Info    | <--------------------------------------------------------->
Rank:        0 | Info    | <     Start calc_SeisSol ...                              >
Rank:        0 | Info    | <--------------------------------------------------------->
Wed Jan 18 10:31:04, Info:  Waiting for last wave field.
Wed Jan 18 10:31:04, Info:  Writing wave field at time 0.
Wed Jan 18 10:31:05, Info:  Writing wave field at time 0. Done.
Wed Jan 18 10:31:05, Info:  Writing free surface at time 0.
Wed Jan 18 10:31:06, Info:  Writing free surface at time 0. Done.
SeisSol_Debug_dskx_6_elastic: /dss/dsshome1/03/ga35kum3/SeisSol/master/src/Solver/time_stepping/GhostTimeCluster.cpp:26: void seissol::time_stepping::GhostTimeCluster::receiveGhostLayer(): Assertion `ct.predictionTime > lastSendTime' failed.
SeisSol_Debug_dskx_6_elastic: /dss/dsshome1/03/ga35kum3/SeisSol/master/src/Solver/time_stepping/GhostTimeCluster.cpp:26: void seissol::time_stepping::GhostTimeCluster::receiveGhostLayer(): Assertion `ct.predictionTime > lastSendTime' failed.
SeisSol_Debug_dskx_6_elastic: /dss/dsshome1/03/ga35kum3/SeisSol/master/src/Solver/time_stepping/GhostTimeCluster.cpp:26: void seissol::time_stepping::GhostTimeCluster::receiveGhostLayer(): Assertion `ct.predictionTime > lastSendTime' failed.
SeisSol_Debug_dskx_6_elastic: /dss/dsshome1/03/ga35kum3/SeisSol/master/src/Solver/time_stepping/GhostTimeCluster.cpp:26: void seissol::time_stepping::GhostTimeCluster::receiveGhostLayer(): Assertion `ct.predictionTime > lastSendTime' failed.

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 1 PID 3718051 RUNNING AT i05r06c03s05
=   KILLED BY SIGNAL: 6 (Aborted)
===================================================================================
```

**Additional context**

**Comment by davschneller** [comment]
If it helps to note: I have only encountered the assertion so far if (and only if) `COMMTHREAD=ON`. Once the communication thread was off, the assertion didn't cause problems anymore.

**Comment by krenzland** [comment]
It should probably be `ct.predictionTime >= lastSendTime`
I don't think that this is actually just an incorrect assertion and not a bug in the scheduling

**Comment by davschneller** [comment]
Fixed by #847.

(cf. https://github.com/SeisSol/SeisSol/pull/847#discussion_r1183089180)
